src/scmdata/ops.py

```diff
@@ -557,7 +557,61 @@ def divide(self, other, op_cols, **kwargs):
     return type(self)(out)
 
 
-def integrate(self, out_var=None):
+def _integrate_sum(emissions):
+    time_unit = "a"
+
+    # Check that all intervals are uniform and equal
+    years = emissions["year"]
+    if not (years.diff().iloc[1:] == 1).all():
+        raise ValueError(
+            'Annual data are required for "sum" integration. See ScmRun.resample'
+        )
+
+    ts = emissions.timeseries()
+
+    # Find the first continuous set of valid values
```

Comment:
Does https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.cumsum.html with `skipna=False` not do this for us (so we can avoid the loop over rows)?

Reply:
That could break the assumption that data are on annual timesteps. With the current implementation, any values after a nan (excluding leading nans) are nan. AFAIK skipna would effectively assume that the missing year is 0

Reply:
I think if `skipna = False` then nans are included in the sum i.e. any values after the sum will be nan (nan + anything = nan). Pandas cumsum would fail the leading nans stuff though but you can also just filter to remove leading nans. Happy to leave but it seems like this loop could be super slow on a big dataset.

Reply:
After some testing pd.DataFrame.cumsum works and is more inline with the trapz implementation.

Including the `reference_year` arg solves the issue with leading nans that the previous implementation was trying to work around

Reply:
I still don't completely see the need for `reference_year`. Either you filter first or use `relative_to_ref_period_mean` after (and with the convention of up to and including, you just have to do relative to one year earlier). I am happy to roll over on this though.

Reply:
I pulled it out. Can add back in a later date. I don't really like the `year - 1` approach as I find that a bit confusing, I'll just filter manually